test/db/e2e/metamodel_test_impl.dart

```diff
@@ -79,7 +80,7 @@ runTests(datastore, db.DatastoreDB store) {
               }
               var partition = store.newPartition(namespace.name);
               var kindQuery = store.query(Kind, partition: partition);
-              futures.add(kindQuery.run().toList().then((List<Kind> kinds) {
+              futures.add(kindQuery.run().toList().then((List<db.Model> kinds) {
```

Comment:
The resulting `kinds` will be of type `Kind`, you might want to cast the iterable/list instead of changing the type to `db.Model`.

Reply:
True, but the call to `cond` doesn't forward any type information – so the `k.name` call is dynamic.

This just eliminates a fuzzy arrow hint – which will be a warning soon in Dart v2.

The long-term solution here is to make `Query` generic on the model type returned.